packages/gatsby-remark-embed-snippet/src/index.js

```diff
@@ -73,6 +99,25 @@ module.exports = ({ markdownAST, markdownNode }, { directory } = {}) => {
           .split(`\n`)
           .filter((_, lineNumber) => lines.includes(lineNumber + 1))
           .join(`\n`)
+      } else if (sname.length) {
+        let index1 = code.indexOf(`start-snippet{${sname}}`)
+        if (index1 > -1) {
+          let index2 = code.indexOf(`\n`, index1)
+          if (index2 > -1) {
+            index2 = index2 + 1 // skip the newline
```

Comment:
I didn't want the newline at the end of the comment containing `start-snippet` to be included in the included code.

Reply:
Ah, right - this makes sense.

I do wonder if we could make the code more concise (and drop `index1` up to `index4` which makes it bit hard to follow) in the process:

what instead of using `(last)indexOf` we would use regular expressions:

```js
const startingSnippetDirectiveMatcher = new RegExp(`start-snippet{${sname}}[^\r\n]*[\r\n](.*)`, `gs`)
const startSnippetMatch = startingSnippetDirectiveMatcher.exec(code)
if (startSnippetMatch?.length >= 2) {
  code =  startSnippetMatch[1]
}

const endingSnippetDirectiveMatcher = new RegExp(`(.*)[\r\n][^\r\n]*end-snippet{${sname}}`, `gs`)
const endSnippetMatch = endingSnippetDirectiveMatcher.exec(code)
if (endSnippetMatch?.length >= 2) {
  code =  endSnippetMatch[1]
}
```

This would pick next lines after `start-snippet{name}` in first step and lines before `end-snippet` in the second.

Alternative is to keep current code - just move it to some utility file to not show all the "gruesome" details of extracting snippets in main file which should be bit more high level than details like.

I'm fine with both approaches

Reply:
I like the regular expressions - used that approach. Thanks for the suggestion!